irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -86,8 +87,17 @@ grpc::Status MstTransportGrpc::SendState(
   async_call_->log_->info("batches in MstState: {}",
                           new_state.getBatches().size());
 
+  shared_model::crypto::PublicKey source_key(request->source_peer_key());
+  auto key_invalid_reason =
+      shared_model::validation::validatePubkey(source_key);
+  if (key_invalid_reason) {
+    async_call_->log_->info(
+        "Dropping received MST State due to invalid public key: {}",
+        *key_invalid_reason);
+  }
+
   subscriber_.lock()->onNewState(
```

Comment:
Seems that this line could be executed after the fail of `request::source_peer_key` validation.